omegaconf/basecontainer.py

```diff
@@ -433,6 +434,8 @@ def _set_item_impl(self, key: Any, value: Any) -> None:
 
         from .nodes import AnyNode, ValueNode
 
+        if isinstance(value, AnyNode):
+            value = value._value()
```

Comment:
Please explain why you need it, it seems out of place.
The following block is differentiating between Nodes (isintance(value, Node)) and Containers (the else branch).
Your change seems to be causing AnyNodes to be treated as containers. I don't understand why it's needed.

Reply:
I don't like this too, I'm going to look for a better way to deal with this. 
I made this hacky fix in order to deal with merge with a Config like `OmegaConf.create({"a": 4})` since `4` will be AnyNode and it cause to not check the value correctly.